pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -126,6 +131,26 @@ func moveVFtoContainerNS(vfNetDevice string, containerID string, containerNetNS
 			return fmt.Errorf("failed to configure IP address for container %s: %v", containerID, err)
 		}
 		klog.V(2).Infof("ipam.ConfigureIface result: %+v, err: %v", result, err)
+		// For VF interface in a Pod or container namespace (without VF Representator), configure policy rules to isolate the route lookup specific to the secondary network.
+		for _, ip := range result.IPs {
+			updateIPRule := netlink.NewRule()
+			updateIPRule.Table = ipRouteCustomTableID
+			updateIPRule.Src = &ip.Address
+			updateIPRule.Priority = ipRouteCustomTablePri
+			// Configure ip rule specific to the secondary network subnet.
+			if err := netlink.RuleAdd(updateIPRule); err != nil {
+				return fmt.Errorf("failed to add IP rule for %s: %v. moveVFtoContainerNS terminated.", containerIfaceName, err)
+			}
+
+			_, ipNetwork, _ := net.ParseCIDR(string(ip.Address.IP))
+			link, err = netlink.LinkByName(containerIface.Name)
+			defaultRoute := &netlink.Route{LinkIndex: link.Attrs().Index, Dst: ipNetwork, Src: ip.Address.IP, Gw: net.IPv4(0, 0, 0, 0), Table: ipRouteCustomTableID,
+				                       Scope: netlink.SCOPE_LINK}
+			// Add a defaultroute to route the secondary network specific traffic through the VF interface directly.
```

Comment:
But you do not configure a gateway for secondary network, and so I think even without the ip rule and route, traffic to the secondary network subnet will still go from the secondary interface anyway, no?

Reply:
indeed yes. But, I Introduced a separate policy route table to isolate the secondary network lookup. But, for the current implementation with SRIOV, this is not required. I have removed it and let the routing b/w VFs handled with the main table entry itself. (we need to introduce this back with veth pair).